frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-root-page/segment-root-page-content/segment-global-section-card/segment-global-section-card-table/segment-global-section-card-table.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Observable } from 'rxjs';
+
+import { ChangeDetectionStrategy, Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+
+import { MatTableDataSource, MatTableModule } from '@angular/material/table';
+import { AsyncPipe, NgIf, NgFor } from '@angular/common';
+import { RouterModule } from '@angular/router';
+import { MatSort } from '@angular/material/sort';
+import { CommonStatusIndicatorChipComponent } from '../../../../../../../../shared-standalone-component-lib/components';
+import { SegmentsService } from '../../../../../../../../core/segments/segments.service';
+import { SharedModule } from '../../../../../../../../shared/shared.module';
+import {
+  Segment,
+  SEGMENT_ROOT_COLUMN_NAMES,
+  SEGMENT_ROOT_DISPLAYED_COLUMNS,
+  SEGMENT_TRANSLATION_KEYS,
+} from '../../../../../../../../core/segments/store/segments.model';
+
+@Component({
+  selector: 'app-segment-global-section-card-table',
+  imports: [MatTableModule, AsyncPipe, NgIf, NgFor, SharedModule, RouterModule, CommonStatusIndicatorChipComponent],
+  templateUrl: './segment-global-section-card-table.component.html',
+  styleUrl:
+    '../../segment-root-section-card/segment-root-section-card-table/segment-root-section-card-table.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SegmentGlobalSectionCardTableComponent implements OnInit {
+  @Input() dataSource$: MatTableDataSource<Segment>;
+  @Input() isLoading$: Observable<boolean>;
+  segmentSortKey$ = this.segmentsService.selectGlobalSegmentSortKey$;
+  segmentSortAs$ = this.segmentsService.selectGlobalSegmentSortAs$;
+
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
+  @ViewChild('tableContainer') tableContainer: ElementRef;
+
+  constructor(private segmentsService: SegmentsService) {}
+
+  ngOnInit() {
+    this.sortTable();
+  }
+
+  ngOnChanges() {
+    this.sortTable();
+  }
+
+  private sortTable() {
+    if (this.dataSource$?.data) {
+      this.dataSource$.sortingDataAccessor = (item, property) =>
+        property === 'name' ? item.name.toLowerCase() : item[property];
+      this.dataSource$.sort = this.sort;
+    }
+  }
+
+  get displayedColumns(): string[] {
+    return SEGMENT_ROOT_DISPLAYED_COLUMNS;
+  }
+
+  get SEGMENT_TRANSLATION_KEYS() {
+    return SEGMENT_TRANSLATION_KEYS;
+  }
+
+  get SEGMENT_ROOT_COLUMN_NAMES() {
+    return SEGMENT_ROOT_COLUMN_NAMES;
+  }
+
+  changeSorting(event) {
```

Comment:
I don't know if others have the same mania I do for slaying redundant code, `changeSorting` and `sortTable` are implemented multiple times the same way, these can be made `common-table-helper` utilities or some other shared way to reduce it at least in the 'refreshed' features.

there's a lot of copy pasta around table functionality though, not saying that needs to be a part of this PR. it may just be too baked in this way to be worth refactoring now, but I'd like to create a ticket for a rainy day.

Reply:
Yeah, I considered trying to abstract the sorting (since we didn't want search here too), but I concluded that it probably would have required a lot of changes in a lot of places.